packages/column-views/src/maps/column-navigation/mapbox/state.ts

```diff
@@ -76,13 +77,15 @@ export function ColumnNavigationProvider({
   //   store.setState({ columns: _columns, selectedColumn });
   // }, [projectID, inProcess, columns, getColumns]);
 
-  const _columns = columns ?? useMacrostratColumns(projectID, inProcess);
+  const newColumns = getColsData({columns});
+  const _columns = columns ? newColumns : useMacrostratColumns(projectID, inProcess);
+
```

Comment:
Instead of getting macrostrat columns separately, just filter the existing columns on the client side. Right now you are getting both, which is unnecessary.